### DISCUSSION TITLE
Wildfires

### DISCUSSION THREAD

**Discussion by Volte6** [GoMudEngine/GoMud#88] [Q&A]
Toying with the idea of wildfires... basically works like this. Curious what other similar effects should be supported, especially if they can spread from area to area. If there aren't any good ones, I'll probably just leave it as a one-off feature, but I'll code it to be generic if there are other uses for this sort of thing (Room effect which spreads to other rooms)

<img width="538" alt="image" src="https://github.com/user-attachments/assets/422d5b83-1d98-4e99-8fef-d06d9d5f1981">

**Comment by ArmoredArtist** [comment]
It would be interesting to see it grow then shrink back down to the starting room and go out dynamically. A smoke effect that varies during and after it goes out that could affect endurance, round times, visibility or another stat of your choice might add more depth. Just random ideas that came to mind. Looks cool.

**Comment by ArmoredArtist** [comment]
If you have crafting in the game, this would be a good excuse to create a window to collect coal for the player. It would give a player a reason to actually seek out this event as well.

**Comment by Volte6** [comment]
Okay, so essentially a room effects system of some sort... I'll think on that a bit and probably make it more generic. A room based effect system could definitely have utility, from weather (rain, fire, snow, fog, smoke) to other less obvious things like avalanches. I'll need to either define these effects in flat files/scripts or write native go code for them... i do prefer to make them scriptable and defined by flat files but it's more work, so I may forego that until I feel that the need is proven.

Re crafting: I have notes on a potential crafting system here if you want to weigh in: https://github.com/Volte6/GoMud/discussions/52

I haven't started on it, just letting it percolate for a bit until I feel confident in an approach, since that one feels like a bit more work than other minor features.

**Comment by Volte6** [comment]
This is in for now, with wildfires working int he forest... currently really only happen if you toss a grenade in the forest.

Closing this, but will feel out the feature and see if it's worth keeping.
